stable/database/files/nuobackup

```diff
@@ -175,82 +172,28 @@ if [ -n "$semaphore" ]; then
 
 
    if [ -z "$gate" ]; then
-      echo >&2 "Semaphore $semaphore/$db_name empty - nothing to do"
+      echo "Semaphore $semaphore/$db_name empty - nothing to do"
       exit 0
    fi
 
    # wait for the database to be RUNNING
    nuocmd check database --db-name $db_name --check-running --wait-forever
 fi
 
-# find the index for the latest backup info
-latest=$(nuocmd get value --key $NUODB_BACKUP_KEY/$db_name/${backup_group}/latest )
-current=$latest
-
-if [ "$backup_type" = "full" ]; then
-   if [ -z "$latest" ]; then
-      current=0
-   fi
-
-   # bump latest, and wrap around (simple ring-buffer)
-   next=$(( (current + 1) % $NUODB_MAX_BACKUP_HISTORY ))
-else
-   if [ -z "$latest" ]; then
-      echo >&2 "No full backup has completed yet - try again later"
-      exit -1
+if [ "$backup_type" != "full" ]; then
+   error="$( nuodocker get current-backup \
+      --db-name "${db_name}" \
+      --labels "backup ${backup_group} ${labels}" 2>&1)"
+   if [ $? -ne 0 ] && echo "$error" | grep -q "No existing backup path found"; then
+      # Check if full backup has been started already for all archiveIds
+      # This will most likely execute in the following scenarious:
+      # * initial database backup request
+      # * backup request after the database have been restored
```

Comment:
I see the comment - but I don't yet understand how this works after a database has been restored.
Why does the `nuodocker get current-backup` fail after the database has been restored?

Reply:
If `get current-backup` is not able to find the backupset name for some `archiveId`, it fails with "No existing backup path found for archive id x".
This can be used is several use cases:
* after restore - The restoration process involves the creation of new archives with new `archiveIds`. 
* after HC SM scale-up - The new `archiveId` will be detected.

Reply:
Got it!
THanks!